test_scripts/ATF_ShutDown_MASTER_RESET.lua

```diff
@@ -0,0 +1,149 @@
+  commonFunctions:waitForSDLStart(self):Do(function()
+    self:initHMI():Do(function()
+      commonFunctions:userPrint(35, "HMI initialized")
+      self:initHMI_onReady():Do(function ()
+        commonFunctions:userPrint(35, "HMI is ready")
+        self:connectMobile():Do(function ()
+          commonFunctions:userPrint(35, "Mobile Connected")
+          self:startSession():Do(function ()
+            commonFunctions:userPrint(35, "App is registered")
+          end)
+        end)
+      end)
+    end)
+  end)
+end
+
+function Test:Activate_App_And_Put_File()
+  commonSteps:ActivateAppInSpecificLevel(self, self.applications["Test Application"])
+  local on_hmi_full = EXPECT_NOTIFICATION("OnHMIStatus", {hmiLevel = "FULL", systemContext = "MAIN"})
+
+  local function addCommand()
+    local cid = self.mobileSession:SendRPC("AddCommand",{ cmdID = 1005,
+              vrCommands = { "OnlyVRCommand"}
+            })
+    EXPECT_HMICALL("VR.AddCommand", {cmdID = 1005, type = "Command",
+                   vrCommands = {"OnlyVRCommand"}}):Do(function(_,data)
+      self.hmiConnection:SendResponse(data.id, data.method, "SUCCESS", {})
+    end)
+    EXPECT_RESPONSE(cid, { success = true, resultCode = "SUCCESS" })
+    EXPECT_NOTIFICATION("OnHashChange"):Do(function(_, data)
+      hash_id = data.payload.hashID
+    end)
+  end
+
+  on_hmi_full:Do(function()
+    local cid = self.mobileSession:SendRPC(
+    "PutFile", {
+      syncFileName = "icon.png",
+      fileType = "GRAPHIC_PNG",
+      persistentFile = true,
+      systemFile = false,
+    }, "files/icon.png")
+    EXPECT_RESPONSE(cid, { success = true, resultCode = "SUCCESS" })
+    addCommand()
+  end)
+end
+
+--[[ Test ]]
+commonFunctions:newTestCasesGroup("Check that SDL finish it's work properly by MASTER_RESET")
+
+function Test:ShutDown_MASTER_RESET()
+  self.hmiConnection:SendNotification("BasicCommunication.OnExitAllApplications", 
+    { reason = "MASTER_RESET" })
+  EXPECT_NOTIFICATION("OnAppInterfaceUnregistered", { reason = "MASTER_RESET" })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnAppUnregistered", { unexpectedDisconnect = false })
+  EXPECT_HMINOTIFICATION("BasicCommunication.OnSDLClose"):Do(function ()
+    StopSDL()
+  end)
+end
+
+--- Start SDL again then add mobile session
+function Test:Restart_SDL_And_Add_Mobile_Connection()
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self):Do(function()
+    self:initHMI():Do(function()
+      commonFunctions:userPrint(35, "HMI initialized")
+      self:initHMI_onReady():Do(function ()
+        commonFunctions:userPrint(35, "HMI is ready")
+        self:connectMobile():Do(function ()
+          commonFunctions:userPrint(35, "Mobile Connected")
+        end)
+      end)
+    end)
+  end)
+end
+
+--- Check SDL will not resume application when the same application registers.
+function Test:Check_Application_Not_Resume_When_Register_Again()
+  local mobileSession = mobile_session.MobileSession(self, self.mobileConnection)
```

Comment:
@LitvinenkoIra , inconsistent style of var names. Please choose one and follow. I suggest "mobileSession" -> "mobile_session "